Assets/MixedRealityToolkit.Tests/EditModeTests/Core/MixedRealityToolkitFilesTests.cs

```diff
@@ -123,6 +154,54 @@ public void TestValidPath(string path, MixedRealityToolkitModuleType expectedMod
         public void CleanupTests()
         {
             TestUtilities.EditorTearDownScenes();
+            if (adhocTestDirectoryCreated)
+            {
+                RecursiveFolderCleanup(new DirectoryInfo(Path.Combine(Application.dataPath, "lib")));
+                adhocTestDirectoryCreated = false;
+            }
+        }
+
+        /// <summary>
+        /// Creates a directory under the Assets path that looks like:
+        /// Assets/lib/mrtk/MixedRealityToolkit.Adhoc/empty
+        /// </summary>
+        /// <remarks>
+        /// This must be cleaned up in CleanupTests.
+        /// </remarks>
+        private void CreateAdhocTestingPackageDirectory()
+        {
+            adhocTestDirectoryCreated = true;
+            Directory.CreateDirectory(Path.Combine(Application.dataPath, "lib"));
+            Directory.CreateDirectory(Path.Combine(Application.dataPath, "lib/mrtk"));
+            Directory.CreateDirectory(Path.Combine(Application.dataPath, "lib/mrtk/MixedRealityToolkit.AdhocTesting"));
+            Directory.CreateDirectory(Path.Combine(Application.dataPath, "lib/mrtk/MixedRealityToolkit.AdhocTesting/empty"));
+        }
+
+        private static void RecursiveFolderCleanup(DirectoryInfo folder)
+        {
+            foreach (DirectoryInfo subFolder in folder.GetDirectories())
```

Comment:
Why not just call folder.Delete(true) and then you just need to delete the associated meta file?

If you are worried about re-use of folder by other components, add a random guid?

Reply:
Actually does the current code delete lib.meta?

Reply:
This code itself doesn't, but the part in TestCleanup does - FWIW, I took this from the MSBuild code that was added here (I guess that @andreiborodin tried a few different things here and this is what ended up working)

I didn't choose to make it common because of the direction where we'll be moving the msbuild code to a different repo in the medium term, and didn't want to add more things to disentangle

Reply:
I'm slightly confused. Test cleanup calls RecursiveFolderCleanup("...\Assets\lib") but I don't think this actually deletes lib.meta?

I'm not saying necessarily make RecursiveFolderCleanup() in a common place like Utils. I am asking why can't the function be simplified to like the below? Simpler to read & maintain

CleanupFolder(path)
{
// if dir exists path
var dir = new DirectoryInfo(path);
dir.Delete(true);

// if path + ".meta" file exists
var metaFile = new FileINfo(path);
file.delete();
}